include/openthread/thread.h

```diff
@@ -500,6 +500,18 @@ const otIp6Address *otThreadGetLinkLocalAllThreadNodesMulticastAddress(otInstanc
  */
 const otIp6Address *otThreadGetRealmLocalAllThreadNodesMulticastAddress(otInstance *aInstance);
 
+/**
+ * This function retrieves the Service ALOC for given Service ID.
+ *
+ * @param[in]   aInstance     A pointer to an OpenThread instance.
+ * @param[in]   aServiceId    Service ID to get ALOC for.
+ * @param[out]  aServiceAloc  A pointer to output the Service ALOC. MUST NOT BE NULL.
+ *
+ * @retval OT_ERROR_NONE      Successfully retrieved the Service ALOC.
+ * @retval OT_ERROR_DETACHED  The Thread interface is not currently attached to a Thread Partition.
+ */
+otError otThreadGetServiceAloc(otInstance *aInstance, uint8_t aServiceId, otIp6Address *aServiceAloc);
```

Comment:
The "API version" github action check is failing. Since this PR is adding a new public API, we need to also update the `OPENTHREAD_API_VERSION` in `include/openthread/instance.h` header file.